dvc/command/diff.py

```diff
@@ -157,13 +157,13 @@ def add_parser(subparsers, parent_parser):
         ),
     )
     diff_parser.add_argument(
-        "a_ref", help="Git reference from which diff calculates"
+        "a_ref", help="Git reference from which the diff begins"
     )
     diff_parser.add_argument(
         "b_ref",
         help=(
-            "Git reference until which diff calculates, if omitted "
-            "diff shows the difference between current HEAD and a_ref"
+            "Git reference until which the diff ends. "
+            "If omitted, `HEAD` (latest commit) is used."
         ),
```

Comment:
@iterative/engineering should we call these "commits" instead of "references"? See https://git-scm.com/docs/git-diff for example. Note that a `ref` in Git is something else: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefrefaref – a bit more technical.

Note that per https://github.com/iterative/dvc.org/pull/933#pullrequestreview-348855380, in docs we're going to start using the term "revision" (link to https://git-scm.com/docs/revisions) which according to [its glossary entry](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefrevisionarevision) is a "synonym of commit (the noun)" so you we could just say "Git revision" here instead, so we use that term consistently everywhere. (never "reference").

Thoughts?

Reply:
I went for this criteria in iterative/dvc.org/pull/933: Use "version" for data or DVC-tracked files, "revision" (and that link) in the context of pure Git/ source code, and "reference" sparingly when actually talking about Git references (to revisions/commits). Updating this repo accordingly...

p.s. We're also reviewing the use of "commit hash" and "SHA" terms...

Reply:
UPDATE: Honestly, I think it would be nicer to only use one single term in the code base. In this case "revision" seems the most appropriate, so these params could be `a_rev` and `b_rev`. I would vote for "revision" because we already use it extensively for the `dvc get` and `dvc import` commands, in the form of the `--rev` option as well as `rev` and `rev_lock` fields of import stage DVC-files.

> Another term that may need review is "checksum", per https://github.com/iterative/dvc.org/pull/933#pullrequestreview-350538870.

Reply:
that's the best way to explain it https://git-scm.com/docs/git-diff - ` Show changes between commits, commit and working tree, etc` .. just plain commits! I would vote to go with just commits everywhere. It would simplify the whole thing - understanding param names, docs, etc. Of course we can put a link to the ways of specifying commits.

Reply:
if not commits, let's at least use revisions indeed. To at least have consistency with `dvc get`\`dvc import` and at least use the correct term.

Reply:
@jorgeorpinel , @shcheklein I recently opened an issue about exactly this :sweat_smile:  #3245 

I would prefer to use either revisions or commits. Revisions is more accurate but _commits_ are more user friendly, I guess both newbies and advanced users understand what would be talking about.

Reply:
Commit is accurate. If you examine https://git-scm.com/docs/gitglossary#def_revision, "revision" = "Synonym for commit (the noun)." 🙂 

> I recently opened an issue about exactly this 😅 #3245

Wow! OK, moving this discussion over there and resolving here.

p.s. please someone review this PR when you can 😬